api/types/mfa.go

```diff
@@ -0,0 +1,17 @@
+package types
+
+import "github.com/gravitational/trace"
+
+func (d *MFADevice) HideSecrets() error {
```

Comment:
If you want to remove error, what should happen in case there is not supported device type? I am afraid that if we just remove it, someone will forget to add new case in WithoutSensitiveData when new type will be added. 
what do you think @codingllama ?

Reply:
Fair enough - we've added device types before, so it might happen again.

WDYT? `func (d *MFADevice) WithoutSensitiveData() (*MFADevice, error)`?

(In hindsight the name is a bit long, but nothing better occurs to me right now.)

Reply:
done